CMakeLists.txt

```diff
@@ -104,10 +104,6 @@ if (${CMAKE_COMPILER_IS_GNUCXX})
 
     add_compile_options(
         -Werror=return-type      # Error out if function definitions lack a return type
```

Comment:
The temporary fix https://github.com/supercollider/supercollider/pull/6436/files also turned `-Wreturn-type` into `-Werror=return-type`.
Should this be reverted as well?

Otherwise this seems fine to me, GCC14 pipeline runs.

Reply:
>Should this be reverted as well?

It seems that this is unrelated to the temporary fix, so we might keep this. (maybe we can ask Spacechild1?)

[quote](https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wreturn-type) from GCC doc:
>-Wreturn-type

>    Warn whenever a function is defined with a return type that defaults to int (unless -Wimplicit-int is active, which takes precedence). Also warn if execution may reach the end of the function body, or if the function does not contain any return statement at all. 

and `Werror=return-type` will make the `return-type` warning into an error.

Reply:
Yes, we should definitely keep this! I once spend half a day trying to fix a bug before I realized I forgot the return statement in a method. From this day, the first thing I do in every new project is `-Werror=return-type`. (I don't understand why this still isn't the default...)